localstack-core/localstack/testing/testselection/scripts/generate_test_selection_from_pr.py

```diff
@@ -19,19 +24,28 @@ def generate_from_pr(
     matching_rules: list[MatchingRule] | None = None,
     repo_name: str = "localstack",
 ):
-    if len(sys.argv) != 4:
-        print(
-            f"Usage: python -m {repo_name}.testing.testselection.scripts.generate_test_selection_from_pr <git-root-dir> <pull-request-url> <output-file-path>",
-            file=sys.stderr,
-        )
-        sys.exit(1)
-
-    output_file_path = sys.argv[-1]
-    pull_request_url = sys.argv[-2]
-    repo_root_path = sys.argv[-3]
+    parser = argparse.ArgumentParser(description="Generate test selection from a pull request")
+    parser.add_argument("git_root_dir", type=str, help="Path to the git repository root")
+    parser.add_argument("output_file_path", type=str, help="Path to the output file")
+    parser.add_argument("--pr", type=str, help="Pull request URL")
+    args = parser.parse_args()
 
+    output_file_path = args.output_file_path
+    repo_root_path = args.git_root_dir
     github_token = os.environ.get("GITHUB_API_TOKEN")
 
+    if args.pr is None:
+        current_branch = get_branch_name(repo_root_path)
+        print(
+            f"No pull request URL provided, evaluating based on current branch ({current_branch})"
+        )
+        pull_request_url = get_pr_url_from_branch(
+            f"localstack/{repo_name.replace('_', '-')}", current_branch, token=github_token
+        )
+        print(f"Detected pull request: {pull_request_url}")
```

Comment:
I feel like this could maybe be a bit simplified. If I got this right, this function tries to find the PR URL (by using the search API) such that it is parsed later (extract the number from the URL) to use the API (with the number) to fetch the SHA of the head and the base (which might already be in the response of the very first request)?

Reply:
Yes, you can get the base and head sha already when receiving the PR url. I didn't want to disrupt the existing flow too much. But I can refactor it a bit so there are no unnecessary http requests

Reply:
Please do so, this code looks like it's just patched in to get things out, but there's no need to be hectic here. Let's make sure that the code here gets cleaner and more maintainable (instead of the other way around 😛).